addons/mrp/static/src/widgets/timer.js

```diff
@@ -0,0 +1,66 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+const { Component, useState, onWillUpdateProps, onWillStart, onWillPatch, onWillDestroy } = owl;
+
+export class MrpTimer extends Component {
+    setup() {
+        super.setup();
+        this.state = useState({
+            hours: this.props.hours !== undefined ? this.props.hours : 0,
+            minutes: this.props.minutes !== undefined ? this.props.minutes : 0,
+            seconds: this.props.seconds !== undefined ? this.props.seconds : 0,
+        });
+
+        this.ongoing = this.props.ongoing;
+        if (this.props.record.data) {
+            this.ongoing = this.props.record.data.is_user_working;
+            this.state.seconds = this.props.record.data.duration * 60;
+        }
+        this._updateTimer();
+        onWillStart(() => this._runTimer());
+        onWillUpdateProps((nextProps) => {
+            this.ongoing = nextProps.ongoing;
+            this._runTimer();
+        });
+        onWillPatch(() => this._updateTimer());
+        onWillDestroy(() => clearTimeout(this.timer));
+    }
+
+    get toString() {
+        return _.str.sprintf(
+            "%02d:%02d:%02d",
+            this.state.hours,
+            this.state.minutes,
+            this.state.seconds
+        );
+    }
+
+    _runTimer() {
+        if (this.ongoing) {
+            this.timer = setTimeout(() => {
+                this.state.seconds += 1;
+                this._runTimer();
+            }, 1000);
+        }
+    }
+
+    _updateTimer() {
+        if (this.state.seconds > 59) {
+            const addtionalMinutes = (this.state.seconds / 60) >> 0;
+            this.state.seconds = this.state.seconds % 60;
+            this.state.minutes += addtionalMinutes;
+        }
+        if (this.state.minutes > 59) {
+            const addtionalHours = (this.state.minutes / 60) >> 0;
+            this.state.minutes = this.state.minutes % 60;
+            this.state.hours += addtionalHours;
```

Comment:
Instead of couting seconds, minutes and hours separatly, shouldn't be simpler to update a single count and then call the formatter `float_time` ?
```javascript
import { registry } from "@web/core/registry";

const formatFloatTime = registry.category("formatters").get('float_time');

...

get toString() {
    return formatFloatTime(this.state.seconds, { displaySeconds: true })
}
```
That way, you can remove `_updateTimer` I think :)